File: frontend/src/model/UtilityBillDTOModel.ts
import {CustomExpenseCategoryForBillDTO} from "./CustomExpenseCategoryForBillDTO";

export type UtilityBillDTOModel = {
    year: number,
    prepaymentMonthly: number,
    customExpenseCategoryDTO: CustomExpenseCategoryForBillDTO[];
    designationOfRealEstate: string,
    genderOfTenant: string,
    firstNameOfTenant: string,
    lastNameOfTenant: string,
    roadOfRealEstate: string,
    houseNumberOfRealEstate: string,
    postCodeOfRealEstate: number,
    locationOfRealEstate: string,
    associatedRealEstate: string
}

File: backend/src/main/java/de/neuefische/backend/model/UtilityBillDTOModel.java
package de.neuefische.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UtilityBillDTOModel {
    private int year;
    private double prepaymentMonthly;
    private List<CustomExpenseCategoryDTO> customExpenseCategoryDTO;
    private String designationOfRealEstate;
    private GenderOfTenant genderOfTenant;
    private String firstNameOfTenant;
    private String lastNameOfTenant;
    private String roadOfRealEstate;
    private String houseNumberOfRealEstate;
    private int postCodeOfRealEstate;
    private String locationOfRealEstate;
    private String associatedRealEstate;
}
